saver.py

```diff
@@ -0,0 +1,44 @@
+from pathlib import Path
+import json
+
+from card import Card
+
+from utils import is_parent
+
+
+class Saver:
+    def save(self, card: Card, dir_path='.'):
+        """Save a given card to a given path."""
+        card_path = self.get_card_path(card, dir_path)
+        metadata_path = card_path / "metadata.json"
+        image_path = card_path / "image.jpg"
+
+        metadata = {
+            "name": card.name,
+            "creator": card.creator,
+            "riddle": card.riddle,
+            "solution": card.solution,
+            "image_path": str(image_path)
+        }
+        with metadata_path.open('w') as f:
+            json.dump(metadata, f)
+
+        card.image.image.save(image_path)
+
+    def get_card_path(self, card: Card, dir_path='.') -> Path:
+        """Get the path to save the card in."""
+
+        # the `is_parent` checks are necessary to protect from creators and names like: '..', '/home', ...
+
+        dir_path = Path(dir_path)
```

Comment:
So it will be the same as the `dir_path` parameter of the `save` method. The `save` methods takes a string because that's how it's specified in the exercise. I think it makes most sense to have `dir_path` as an argument to `__init__`, can I do that?

Reply:
https://github.com/roy-cohen273/cardazim/pull/4/commits/313d6d37471b41570209c92eba9bb965109f524f